resources/tsx/class-likes.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+
+class Person{
+  name: { firstname: string; lastname: string; } | undefined;
+}
+type MyProps = {
+  // using `interface` is also ok
+  message: string;
+};
+type MyState = {
+  count: number; // like this
+};
+class App extends React.Component<MyProps, MyState> {
+  state: MyState = {
+    // optional second annotation for better type inference
+    count: 0,
+  };
+  render() {
+
+    return (//this is React specific
+        // @ts-ignore
+        <div>
+          {this.props.message} {this.state.count}
+        </div>
+    );
+  }
+}
+
+type CustomValue = any;
+interface Props {
+  propA: CustomValue;
+}
+interface DefinedState {
+  otherStateField: string;
+}
+type State = DefinedState & ReturnType<typeof transformPropsToState>;
```

Comment:
Okay, whats that now?

Reply:
that will not be counted :D but i will mention that in the doc at #168. 
But @ce-bo maybe you can decide explicitly if this should be counted :)

Reply:
This is a Union, right? So let's not count this way of defining an Alias .

Reply:
Is that really called a Union? We currently write in our Readme that we do count Unions as classes, because in C++, Unions can have member functions and so on. They are a strange memory saving thing otherwise, were you can have multiple variables that are all stored under the same memory location.

Reply:
Ok, these seem to be called `Intersection Types` in TS, so we should be fine.

Reply:
Ok, there are, of course, also the more commonly used `Union Types` via the `|` operator. Well, we have already decided to add language-specific documentation, so we should document that `Unions` are counted in C and C++, but the `Unions` in TS are something else and we do not count them, I guess? @ce-bo  @mylinhdao

Reply:
Further discussion about this topic is now under #213